src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryLogicManager.CreateProvider.cs

```diff
@@ -0,0 +1,334 @@
+            }
+
+            // Searches for the public MethodInfo from the specified type by the given method name
+            // It is a case-sensitive search
+            MethodInfo method = type.GetMethod(retryMethodName);
+            if (method == null)
+            {
+                throw new InvalidOperationException($"Failed to resolve the '{retryMethodName}' method from `{type.FullName}` type.");
+            }
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> The `{2}` method metadata is extracted from the `{3}` type by `{4}` method name."
+                                                    , s_typeName, methodName, method.Name, type.FullName, retryMethodName);
+
+            if (!typeof(SqlRetryLogicBaseProvider).IsAssignableFrom(method.ReturnType))
+            {
+                throw new InvalidCastException($"Invalid return type; It must be as `{typeof(SqlRetryLogicBaseProvider).FullName}` type.");
+            }
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> The return type of the `{2}.{3}()` method is valid."
+                                                    , s_typeName, methodName, type.FullName, method.Name);
+
+            // Prepare the function parameters values
+            object[] funcParams = PrepareParamValues(method.GetParameters(), option, retryMethodName);
+
+            if(method.IsStatic)
+            {
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Run the static `{2}` method without object creation of `{3}` type.",
+                                                       s_typeName, methodName, method.Name, type.FullName);
+                return method.Invoke(null, funcParams);
+            }
+
+            // Since there is no information about the parameters of the possible constructors, 
+            // the only acceptable constructor is the default constructor.
+            object obj = Activator.CreateInstance(type);
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> An instance of `{2}` type is created to invoke the `{3}` method.",
+                                                   s_typeName, methodName, type.FullName, method.Name);
+            return method.Invoke(obj, funcParams);
+        }
+
+        private static object[] PrepareParamValues(ParameterInfo[] parameterInfos, SqlRetryLogicOption option, string retryMethod)
+        {
+            // The retry method must have at least one `SqlRetryLogicOption`
+            if (parameterInfos.FirstOrDefault(x => x.ParameterType == typeof(SqlRetryLogicOption)) == null)
+            {
+                string message = $"Failed to create {nameof(SqlRetryLogicBaseProvider)} object because of invalid {retryMethod}'s parameters." +
+                    $"{Environment.NewLine}The function must have a paramter of type '{nameof(SqlRetryLogicOption)}'.";
+                throw new InvalidOperationException(message);
```

Comment:
All exception messages need to come from Resource Builder (Strings.resx) and preferably built via SqlUtil.